Python - Update Contact using Python Nylas SDK (#55)
nylas-samples/nylas-hacktoberfest-2022: code-samples/contact/python/update_contact/readme.md

# Update an existing account with Python Nylas SDK

  
  

## Setup

  
  

1) Sync and generate `CLIENT_ID, CLIENT_SECRET and ACCESS_TOKEN` from Nylas Dashboard for the email account you want to process

  

```bash
CLIENT_ID=""
CLIENT_SECRET=""
ACCOUNT_ID=""
```

1) Install dependencies

  

```bash
pip install nylas
pip install python-dotenv
```

## Usage

Execute the python files using command line or in any editor of your choice ensuring python is setup. Update below values in your `.env` file. 

```bash
CLIENT_ID, CLIENT_SECRET and ACCESS_TOKEN 
```
## Pre-requisite 

> Please add the customer id from a contact that was created using Nylas lib in the `.env` file in the below format.

Reviewer: ```suggestion
Please add the contact id from an existing contact in the `.env` file in the below format.
```

```
CONTACT_ID = <contact id>
```

> Read the contact details and fetch a contact ID from here `/code-samples/contact/python/read_contacts/read_contacts.py`  

Reviewer: ```suggestion
Read the contact details and fetch a contact ID using the [read contacts code sample](https://github.com/nylas-samples/nylas-hacktoberfest-2022/tree/main/code-samples/contact/python/read_contacts)
```

'birthday': datetime.date(2014, 6, 1),
'cls': <class 'nylas.client.restful_models.Contact'>,
'company_name': 'Naruto',
'emails': defaultdict(<class 'list'>,
{ 'other': ['grereg-admin@ets.org'],
'personal': ['wxtivmdud@emlhub.com']}),
'given_name': 'Toriko',
'id': '7h9xzwgg7161y25y64nqv2tw',
'im_addresses': defaultdict(<class 'list'>,
{ 'gtalk': ['G', 'o', 'o', 'g', 'l', 'e']}),
'job_title': 'Animator',
'manager_name': '',
'middle_name': 'Tokyo',
'nickname': 'Otaku',
'notes': 'I Love Anime!',
'object': 'contact',
'office_location': 'Japan',
'phone_numbers': defaultdict(<class 'list'>,
{ 'business': ['555 555-5555']}),
'physical_addresses': defaultdict(<class 'list'>,
{ 'work': [ { 'city': 'Tokyo',
'country': 'JP',
'format': 'structured',
'postal_code': '94102',
'secondary_address': None,
'state': 'AB',
'street_address': '12 '
'Daijobu '
'St, '
'8th '
'Floor',
'type': 'work'}]}),
'picture_url': 'https://api.nylas.com/contacts/7h9xzwgg7161y25y64nqv2tw/picture',
'source': 'address_book',
'suffix': 'Shinigami',
'surname': 'Ghoul',
'web_pages': defaultdict(<class 'list'>,
{ 'homepage': ['https://Google.com']})}
```

## Note : Expected Errors

> If you try to update a contact that was not created using a Nylas lib, please expect below error traceback.

Reviewer: ```suggestion
If you try to update a contact that was auto-created using Nylas, you will receive the below error.
```

# Update an existing account with Python Nylas SDK

  
  

## Setup

  
  

1) Sync and generate `CLIENT_ID, CLIENT_SECRET and ACCESS_TOKEN` from Nylas Dashboard for the email account you want to process

  

```bash
CLIENT_ID=""
CLIENT_SECRET=""
ACCOUNT_ID=""
```

1) Install dependencies

  

```bash
pip install nylas
pip install python-dotenv
```

## Usage

Execute the python files using command line or in any editor of your choice ensuring python is setup. Update below values in your `.env` file. 

> 

Reviewer: For usage, include the error if you use an auto-generated contact, recommend the user to use a contact not created through Nylas.
```
nylas.client.errors.NylasApiError: 400 Bad Request. Reason: d5tgr4flai636ool27g5cxagx is an auto-generated contact and this operation isnt supported for auto-generated contacts.. Nylas Error Type: api_error
```

Author: `readme.md` file is updated with the added note.
